op-node/rollup/derive/l1_block_info.go

```diff
@@ -273,8 +273,20 @@ func L1InfoDeposit(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNumber
 	var err error
 	if isEcotoneButNotFirstBlock(rollupCfg, l2BlockTime) {
 		l1BlockInfo.BlobBaseFee = block.BlobBaseFee()
-		l1BlockInfo.BlobBaseFeeScalar = sysCfg.BlobBaseFeeScalar
-		l1BlockInfo.BaseFeeScalar = sysCfg.BaseFeeScalar
+		if l1BlockInfo.BlobBaseFee == nil {
+			// The L2 spec states to use the MIN_BLOB_GASPRICE from EIP-4844 if not yet active on L1.
+			l1BlockInfo.BlobBaseFee = big.NewInt(1)
+		}
+		switch sysCfg.Scalar[0] {
+		case 0:
+			l1BlockInfo.BlobBaseFeeScalar = 0
+			l1BlockInfo.BaseFeeScalar = binary.BigEndian.Uint32(sysCfg.Scalar[28:32])
```

Comment:
For case 0, it needs to be backwards compatible so we don't need to parse a subset of the scalar?

Reply:
We already checked Ecotone is active, and the value needs to fit into the 32 bit destination in the L1 block info. It's fine to enforce the validity here, and can ignore scalar changes that are invalid.